kotlinx-coroutines-test/common/src/TestBuilders.kt

```diff
@@ -269,28 +276,33 @@ public fun runTest(
  *
  * @throws IllegalArgumentException if the [context] is invalid. See the [TestScope] constructor docs for details.
  */
-@ExperimentalCoroutinesApi
+@Deprecated(
+    "Define a total timeout for the whole test instead of using dispatchTimeoutMs. " +
+        "Warning: the proposed replacement is not identical as it uses 'dispatchTimeoutMs' as the timeout for the whole test!",
+    ReplaceWith("runTest(context, timeout = with(kotlin.time.Duration.Companion) { dispatchTimeoutMs.milliseconds }, testBody)"),
```

Comment:
This replacement is way too verbose (also, I doubt it's going to work bug-less in complex contexts).
You probably can replace it (pun intended) with `ReplaceWith("runTest(context, timeout = dispatchTimeoutMs.milliseconds, testBody), imports = [kotlin.time.Duration.Companion]"` (maybe `kotlin.time.Duration.Companion.milliseconds` is required though)